cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,59 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
+ import SignUpPageObject from '../support/pages/signUp.pageObject';
+ import HomePageObject from '../support/pages/home.pageObject';
 
-describe('Sign Up page', () => {
-  before(() => {
+ const signUpPage = new SignUpPageObject();
+ const homePage = new HomePageObject();
+ describe('Sign Up page', () => {
+   let user;
 
-  });
+   before(() => {
+     cy.task('db:clear');
+     cy.task('generateUser').then((generateUser) => {
+       user = generateUser;
+     });
+   });
 
-  it('should ...', () => {
+   beforeEach(() => {
+     signUpPage.visit();
+   });
 
-  });
-});
+   it('should provide an ability to successfull sign up', () => {
+     homePage.fillField('Email-field', user.email);
+     homePage.fillField('Username-field', user.username);
+     homePage.fillField('Password-field', user.password);
+     signUpPage.clickButtn('.btn');
+     signUpPage.assertAllert();
+     homePage.assertHeaderContainUsername(user.username);
+   });
+
+   it('should not signed up user with invalid "Username" field', () => {
+     homePage.fillField('Email-field', user.email);
+     homePage.fillField('Username-field', user.invalidUsername);
+     homePage.fillField('Password-field', user.password);
+     signUpPage.clickButtn('.btn');
```

Comment:
you have to use qa attributes, here its hard to understand which button you have been clicking